### DISCUSSION TITLE
setKey returns undefined

### DISCUSSION THREAD

**Discussion by DanielGennat** [neuefische/questions#221] [Web] [answered]
Hi,
egal, was ich in meine setKey Funktion eingebe, ich bekomme im console.log nur ein undefined ... warum? :)

//index.js
<img width="355" alt="Bildschirmfoto 2022-10-06 um 11 11 12" src="https://user-images.githubusercontent.com/109812125/194273992-943b12e0-59fb-4d64-b8f3-818e577ea991.png">

(Und warum wird hier hinter `setKey` automatisch ein Komma gesetzt?)
//EditTimerChainForm.js
<img width="448" alt="Bildschirmfoto 2022-10-06 um 11 12 42" src="https://user-images.githubusercontent.com/109812125/194274431-1045c39a-126b-4c08-a81b-1bc24899dfb0.png">

Wahrscheinlich ist es ein Fehler bei Durchreichen, aber ich finde ihn nicht.

[PullRequest](https://github.com/DanielGennat/capstone-project/pull/12)

**Comment by NikRoe** [comment, marked as answer]
Hey Daniel,
you need to rename the variable `key` to something different than `key`. This is a term that is [reserved by React](https://reactjs.org/docs/lists-and-keys.html#keys) and cannot be passed as a prop. You should see an error message describing this issue in your dev tools.

Hope this helps :)

**Reply by DanielGennat** [reply]
ja, jetzt sehe ich sie, danke :)

**Reply by DanielGennat** [reply]
... jetzt bekomme ich eine Fehlermeldung in der `CountdownCircleTimerFunction.js`
![Bildschirmfoto 2022-10-06 um 11 55 18](https://user-images.githubusercontent.com/109812125/194284222-edfe6d39-cc06-4ea1-ab32-b093274249ca.png)

**Reply by DanielGennat** [reply]
habe ihn gefunden. Habe beim Weiterreichen nur die rechte Seite umbenannt:
`key={countdownKey}` -> `countdownKey={countdownKey}`
